test/main.cpp

```diff
@@ -54,20 +54,19 @@ Distributed under the Boost Software License, Version 1.0.
     retcode = 1;                                                                                                                                               \
   }
 
-#ifdef __GNUC__
+#if defined(__clang__) || defined(_MSC_VER)
+template <class T> struct is_literal_type
+{
+  static constexpr bool value = __is_literal_type(T);
+};
+#elif defined(__GNUC__)
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 template <class T> struct is_literal_type
 {
   static constexpr bool value = std::is_literal_type<T>::value;
 };
 #pragma GCC diagnostic pop
```

Comment:
I referred to this part. If `GCC` with `libc++` is used `defined(__clang__)` should be false, and we would try to use the non-existant `std::is_literal_type<T>::value` or am I missing something?

Reply:
I just checked and libstdc++'s source code actually calls GCC's intrinsic which is... exactly the same form than the others. So maybe we can just use __is_literal_type(T) everywhere

Reply:
okay, I just went through godbolt and tried every compiler in there, EDG, ICC even the random TI DSP thing, and they *all* support __is_literal_type(T) so I'd suggest just using this no matter what.